hq_superset/__init__.py

```diff
@@ -8,15 +8,16 @@ def flask_app_mutator(app):
     # Import the views (which assumes the app is initialized) here
     # return
     from superset.extensions import appbuilder
-
     from . import api, hq_domain, oauth2_server, views
+    from .exceptions import OAuthSessionExpired
 
     appbuilder.add_view(views.HQDatasourceView, 'Update HQ Datasource', menu_cond=lambda *_: False)
     appbuilder.add_view(views.SelectDomainView, 'Select a Domain', menu_cond=lambda *_: False)
     appbuilder.add_api(api.OAuth)
     appbuilder.add_api(api.DataSetChangeAPI)
     oauth2_server.config_oauth2(app)
 
+    app.register_error_handler(OAuthSessionExpired, hq_domain.oauth_session_expired)
```

Comment:
How will this cater to "OAuthError"?

Reply:
Haha, I added the `OAuthError` to be caught but didn't push it. See https://github.com/dimagi/commcare-analytics/pull/84/commits/bae3f7a2f47183399d8ecee24add4d5be54d4681. Thanks for reminder.

Reply:
Thanks. Let me know if you want to test this on staging, I can still replicate it predictably

Reply:
Approved, let me know when deployed to staging

Reply:
@mkangia It's deployed.

Reply:
Great, I can see the message to login again now.

![Screenshot from 2024-12-04 17-09-22](https://github.com/user-attachments/assets/bc310af5-1394-4ba6-a351-34b6cbcb9999)